packages/nuxi/src/cli-wrapper.ts

```diff
@@ -2,39 +2,38 @@
  * This file is used to wrap the CLI entrypoint in a restartable process.
  */
 import { fileURLToPath } from 'node:url'
-import { execa } from 'execa'
-import { EXIT_CODE_RESTART } from './constants'
+import { fork } from 'node:child_process'
+import type { ChildProcess } from 'node:child_process'
 
 const cliEntry = fileURLToPath(new URL('../dist/cli-run.mjs', import.meta.url))
 
-async function startSubprocess (preArgs: string[], postArgs: string[]) {
-  const child = await execa(
-    'node',
-    [
-      ...preArgs,
-      cliEntry,
-      ...postArgs
-    ],
-    {
-      reject: false,
-      stdio: 'inherit',
-      env: {
-        ...process.env,
-        NUXI_CLI_WRAPPER: 'true'
-      }
-    }
-  )
-  if (child.exitCode === EXIT_CODE_RESTART) {
-    await startSubprocess(preArgs, postArgs)
-  } else {
-    process.exit(child.exitCode)
-  }
-}
-
-const args = process.argv.slice(2)
-// only enable wrapper in dev command
-if (args[0] === 'dev') {
-  startSubprocess([], args)
+// Only enable wrapper for nuxt dev command
+if (process.argv[2] === 'dev') {
+  process.env.__CLI_ARGV__ = JSON.stringify(process.argv)
```

Comment:
Would this better to be prefixed with `NUXT` or so?

Reply:
I thought first but since finally dev server logic entirely goes to citty, it wouldn't be prefixed with nuxt (PS: No one should depend on it :D)